src/index.js

```diff
@@ -1,23 +1,23 @@
-import template from 'babel-template';
-import syntax from 'babel-plugin-syntax-dynamic-import';
-
-const buildImport = template(`
+export default ({ template }) => {
+  const buildImport = template(`
   (new Promise((resolve) => {
     require.ensure([], (require) => {
       resolve(require(SOURCE));
     });
   }))
 `);
 
-export default () => ({
-  inherits: syntax,
-
-  visitor: {
-    Import(path) {
-      const newImport = buildImport({
-        SOURCE: path.parentPath.node.arguments,
-      });
-      path.parentPath.replaceWith(newImport);
+  return {
+    manipulateOptions(opts, parserOpts) {
```

Comment:
let's add a comment with the `inherits: syntax` code and instructions to use it instead of this inlined code once this package has dropped babel 6 support?